versioned_docs/version-latest/learn/monitoring.md

```diff
@@ -0,0 +1,723 @@
+sudo cp promtool /usr/local/bin/
+```
+Set permissions:
+```bash
+sudo chown prometheus:prometheus /usr/local/bin/prometheus
+sudo chown prometheus:prometheus /usr/local/bin/promtool
+```
+Copy the `consoles` and `console_libraries` directories to `/etc/prometheus` (folder names can vary)
+```bash
+sudo cp -r consoles /etc/prometheus
+sudo cp -r console_libraries /etc/prometheus
+```
+Set permissions:
+```bash
+sudo chown -R prometheus:prometheus /etc/prometheus/consoles
+sudo chown -R prometheus:prometheus /etc/prometheus/console_libraries
+```
+
+Copy `prometheus.yml` to `/etc/prometheus/`:
+```bash
+sudo cp prometheus.yml /etc/prometheus/
+sudo chown prometheus:prometheus /etc/prometheus/prometheus.yml
+```
+
+
+---
+
+### Prometheus.yml on a Prometheus Server
+(incorporating the job for scraping the Subspace node)
+
+To edit the Prometheus configuration file:
+
+```bash
+sudo vi /etc/prometheus/prometheus.yml
+```
+
+Copy and paste these lines into the file:
+
+```bash
+global:
+  scrape_interval: 15s
+
+scrape_configs:
+  - job_name: 'farmer-1'
+    scrape_interval: 5s
+    static_configs:
+      - targets: ['<farmer-1-ip>:xxxx']
+  - job_name: 'farmer-2'
+    scrape_interval: 5s
+    static_configs:
+      - targets: ['<farmer-2-ip>:xxxx']
+  - job_name: 'farmer-3'
+    scrape_interval: 5s
+    static_configs:
+      - targets: ['<farmer-3-ip>:xxxx']
+# Job for scraping metrics from node
+  - job_name: "subspace"
+    scrape_interval: 5s
+    static_configs:
+      - targets: ['localhost:9615']
+  - job_name: "FarmerServer"
+    static_configs:
+      - targets: ["localhost:9100"]
+
+```
+
+Once modifications are done in the `prometheus.yml` file, remember to reload the service:
+
+```bash
+sudo systemctl daemon-reload
+```
+
+Ensure to provide names for the `job_name`.
+
+Please remember to delete archives and all extracted files from the home directory after distributing all components:
+
+```bash
+cd ~
+rm -r prometheus*
+```
+
+---
+
+Here's a revised version:
+
+---
+
+### Creating a Service for Prometheus
+
+Open the service file using your preferred text editor:
+
+```bash
+sudo vi /etc/systemd/system/prometheus.service
+```
+
+Add the following configuration to the file:
+
+```bash
+[Unit]
+Description=Prometheus
+Wants=network-online.target
+After=network-online.target
+
+[Service]
+User=prometheus
+Group=prometheus
+Type=simple
+ExecStart=/usr/local/bin/prometheus \
+    --config.file /etc/prometheus/prometheus.yml \
+    --storage.tsdb.path /var/lib/prometheus/ \
+    --web.console.templates=/etc/prometheus/consoles \
+    --web.console.libraries=/etc/prometheus/console_libraries
+
+[Install]
+WantedBy=multi-user.target
+```
+
+Remember to reload the daemon after making any service changes:
+
+```bash
+sudo systemctl daemon-reload			
+# Run this whenever any service configuration changes
+```
+
+---
+Here's a revised version for better readability:
+
+---
+
+### UFW Configuration
+
+On the Prometheus Server:
+```bash
```

Comment:
Why are we allowing the farmer servers to access port 9100 on the monitoring machine? The Prometheus server will be hitting the farmer servers to scrape metrics, not the other way round?

Reply:
Good question ..